Create and utilize visit_add_plot_plugin function.

biagas opened the pull request:
### Description

Created `visit_add_plot_plugin` function which has all the CMake logic that used to be written to individual plot's CMakeLists.txt.
It automatically handles all commonly named source files and headers.
Modified GenerateCMake.h to write new logic for plots.
Added `Verbatim` tag for use in .code files for plots that have the need to write specialized logic either before or after the autogenerated CMake logic.

Updated plot .xml files where necessary to remove no-longer-needed information.
Updated plot .code files where needed.
Ran xml2cmake for all plots.

Updated visitpy to use visit_add_library, as the plots depend on visitpy.

### Type of change

* ~~[ ] Bug fix~~
* ~~[ ] New feature~~
* ~~[ ] Documentation update~~
* [X] Other
CMake modernization effort.

### How Has This Been Tested?

Compiled and ran tests on Windows/Linux.

### Checklist:

- [X] I have commented my code where applicable.~~
- ~~[ ] I have updated the release notes.~~
- ~~[ ] I have made corresponding changes to the documentation.~~
- ~~[ ] I have added debugging support to my changes.~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works.~~
- [X] I have confirmed new and existing unit tests pass locally with my changes.~~
- ~~[ ] I have added new baselines for any new tests to the repo.~~
- ~~[ ] I have NOT made any changes to [*protocol* or *public interfaces*][3] in an RC branch.~~

biagas committed:
Create visit_add_plot_plugin macro.

Update GenerateCMake to use the new macro.
Update plot .xml files to remove avtXXXFilter files, since they will be added automatically if present.
Added `Verbatim` tag for xml2cmake in .code files, to add logic `Pre` or `Post` in relation to the automatic code generated by xml2cmake.

biagas committed:
Fix a few glitches with visit_add_plot_plugin.

biagas committed:
Merge branch 'task/biagas/cmake_modernization_with_blt' into task/biagas/add_plot_plugin_macro

biagas committed:
Fix merge issue.

cyrush commented on src/plots/Boundary/CMakeLists.txt:
```diff
@@ -1,95 +1,4 @@
-VISIT_PLUGIN_TARGET_FOLDER(plots Boundary ${INSTALLTARGETS})
 
+visit_add_plot_plugin(
+    PNAME      Boundary)
```
simplifying this is awesome!

cyrush reviewed (approved):
This is a very nice evolution!

brugger1 commented on src/CMake/PluginMacros.cmake:
```diff
@@ -229,3 +229,219 @@ function(CREATE_PLUGIN_DEPENDENCIES target comp type)
+        list(APPEND INSTALLTARGETS ${EParTarget})
+    endif()
+
+    # one of these is not needed for plugin vs install, which one?
```
I noticed this comment with a question. I don't know if this was a reminder of something you still wanted to address or just a comment to a future developer that may touch this code. I don't know the answer to the question.

biagas replied:
Thanks for the reminder, it is something I want to address before this is merged into develop. I will make a note to myself.

biagas committed:
Fix RTD docs warnings.

Remove no-longer-valid literal includes from the CMakeConventions section of the StyleGuide. This will need to be rewritten with the modernization effort is finalized.

Update literal includes from Volume plot in the 'UpdatingPlugins' page.
